### PR TITLE
Enable handling of escaped delimiters

### PR DESCRIPTION
**Pull request by menanteau**

### CODE REVIEW

File: python/HeaderService/hslib_salobj.py [hunk 1]
```diff
@@ -700,3 +701,30 @@ def extract_telemetry_channels(telem, start_collection_event=None,
             channels[name] = c
 
     return channels
+
+
+def split_esc(s, sep=':', esc='\\'):
+
+    '''
+    Split a string using separatior (sep) and escape (esc) character
```
Comment (mareuter) [inline python/HeaderService/hslib_salobj.py]: Typo: should be separator

File: python/HeaderService/hslib_salobj.py [hunk 2]
```diff
@@ -700,3 +701,30 @@ def extract_telemetry_channels(telem, start_collection_event=None,
...
+    # print([re.sub(r'\\(.)','\\1',k) for k in re.split(r'(?<!\\):', s)])
+    # ['OBJECT', '2020-06-16T18:43:55.039', 'OBJECT']
+    #
+    # and:
+    #
+    # s = 'OBJECT:2020-06-16T18\\:43\\:55.039:OBJ\\\\ECT'
+    # print([re.sub(r'\\(.)','\\1',k) for k in re.split(r'(?<!\\):', s)])
+    # ['OBJECT', '2020-06-16T18:43:55.039', 'OBJ\\ECT']
+
+    # The regurlar expresions we'll use
```
Comment (mareuter) [inline python/HeaderService/hslib_salobj.py]: Typo: Should be regular expression

**Review by mareuter — APPROVED** [review]
Couple of comment typos, but otherwise looks good to me.

File: python/HeaderService/hslib_salobj.py (hunk 2 above)
Comment (menanteau) [reply python/HeaderService/hslib_salobj.py]: Fixed.

**Pull request merged**